app/bundles/ProjectBundle/Tests/Functional/Controller/ProjectEntityLookupLimitTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\ProjectBundle\Tests\Functional\Controller;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\EmailBundle\Model\EmailModel;
+use PHPUnit\Framework\Assert;
+
+final class ProjectEntityLookupLimitTest extends MauticMysqlTestCase
+{
+    private const LOOKUP_CHOICE_LIST_URL = '/s/ajax?action=project:getLookupChoiceList';
+
+    /**
+     * Test: AJAX search with a common keyword
+     * Must return exactly 1000 results.
+     */
+    public function testAjaxSearchReturnsExactly1000Results(): void
+    {
+        // Arrange: seed 2000 emails
+        $this->createTestEmails(2000);
+
+        // Act: trigger AJAX lookup endpoint
+        $this->client->request('GET', self::LOOKUP_CHOICE_LIST_URL, [
+            'entityType' => 'email',
+            'filter'     => 'Common',
+        ]);
+
+        $this->assertResponseIsSuccessful();
+        $response = $this->client->getResponse();
+        $decoded  = json_decode($response->getContent(), true);
+        Assert::assertIsArray($decoded, 'Response must be a JSON array.');
+        $count = count($decoded);
+        Assert::assertCount(
+            1000,
+            $decoded,
+            "AJAX autocomplete search should return exactly 1000 results, got {$count}."
+        );
```

Comment:
I think this can be simplified. PHPUNIT will provide the numbers in the failed assertion message.
```suggestion
        Assert::assertCount(
            1000,
            $decoded,
            "AJAX autocomplete search should return exactly 1000 results."
        );
```

Reply:
This is fixed.